onboarding/src/main/kotlin/io/homeassistant/companion/android/onboarding/manualserver/ManualServerScreen.kt

```diff
@@ -0,0 +1,216 @@
+                .padding(contentPadding),
+
+        )
+    }
+}
+
+@Composable
+private fun ManualServerContent(
+    isServerUrlValid: Boolean,
+    onConnectClick: () -> Unit,
+    onServerUrlChange: (String) -> Unit,
+    serverUrl: String,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .padding(horizontal = HASpacing.M)
+            .windowInsetsPadding(WindowInsets.ime)
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(HASpacing.XL),
+    ) {
+        Image(
+            modifier = Modifier
+                .padding(top = HASpacing.XL),
+            imageVector = ImageVector.vectorResource(R.drawable.ic_manual_server),
+            contentDescription = null,
+        )
+        Text(
+            text = stringResource(R.string.manual_server_title),
+            style = HATextStyle.Headline,
+        )
+
+        ServerUrlTextField(
+            serverUrl = serverUrl,
+            onServerUrlChange = onServerUrlChange,
+            isServerUrlValid = isServerUrlValid,
+            onConnectClick = onConnectClick,
+        )
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        HAAccentButton(
+            text = stringResource(R.string.manual_server_connect),
+            onClick = onConnectClick,
+            enabled = isServerUrlValid,
+            modifier = Modifier
+                .padding(vertical = HASpacing.XL),
+        )
+    }
+}
+
+@Composable
+private fun ServerUrlTextField(
+    isServerUrlValid: Boolean,
+    onConnectClick: () -> Unit,
+    onServerUrlChange: (String) -> Unit,
+    serverUrl: String,
+    modifier: Modifier = Modifier,
+) {
+    val focusRequester = remember { FocusRequester() }
+    val isError = serverUrl.isNotBlank() && !isServerUrlValid
+
+    HATextField(
+        value = serverUrl,
+        onValueChange = onServerUrlChange,
+        trailingIcon = {
+            if (serverUrl.isNotEmpty()) {
+                IconButton(onClick = { onServerUrlChange("") }) {
+                    Icon(
+                        imageVector = Icons.Default.Close,
+                        contentDescription = stringResource(R.string.manual_server_clear_url),
+                    )
+                }
+            }
+        },
+        placeholder = {
+            Text(
+                text = "http://homeassistant.local:8123",
+                style = HATextStyle.UserInput,
+                color = LocalHAColorScheme.current.colorOnNeutralNormal,
+            )
+        },
+        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
+        keyboardActions = KeyboardActions(
+            onDone = {
+                if (isServerUrlValid) {
+                    onConnectClick()
+                }
+                // This is going to hide the keyboard in any case
+                defaultKeyboardAction(ImeAction.Done)
```

Comment:
\+ also clear focus from the text input field

Reply:
Also call clearFocus - so when tapping done on the keyboard, the keyboard hides (= default action) and the text field is no longer selected.